### DISCUSSION TITLE
[@types/react-dom] cannot infer type after upgrading to React 18

### DISCUSSION THREAD

**Discussion by ad1992** [DefinitelyTyped/DefinitelyTyped#61403] [Issues with a @types package]
After upgrading to React 18 I am getting the below error when generating the types for the project
<img width="458" alt="Screenshot 2022-07-22 at 12 45 53 PM" src="https://user-images.githubusercontent.com/11256141/180384848-063df368-d2e7-487f-b49f-ac269df5e658.png">

It was working fine prior to React 18. `createPortal` is called from Modal Component and seems like it's not able to infer the types of `createPortal` used  in the `Modal` Component.
Removing `createPortal` works fine.

[a comment by an automated account is not shown]

**Reply by eps1lon** [reply]
Can you reduce this to a complete reproduction (either TypeScript playground or cloneable repository)? It's not clear from just the error what's going on.
